rrmngmnt/foreman_host_wrapper.py

```diff
@@ -0,0 +1,301 @@
+        :type foreman_user: str
+        :param foreman_password: foreman password
+        :type foreman_password: str
+        :param api_version: foreman api version
+        :type api_version: int
+        :param search_attr: search element under collection by given attribute
+        :type search_attr: str
+        """
+        super(ForemanCollection, self).__init__()
+        self.foreman_api = Foreman(
+            url=foreman_url,
+            auth=(foreman_user, foreman_password),
+            api_version=api_version
+        )
+        self.collection_name = collection_name
+        if self.collection_name:
+            self.collection = getattr(self.foreman_api, self.collection_name)
+        if not search_attr:
+            self.search_attr = COLLECTION_D[
+                self.collection_name
+            ][SEARCH_ATTR]
+
+    def get_element_id(self, element_name):
+        """
+        Get element id from foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :return: element id
+        :rtype: str
+        """
+        self.logger.info(
+            "Looking for %s under foreman collection %s",
+            element_name, self.collection_name
+        )
+        response = self.collection.index(
+            search="%s = \"%s\"" % (self.search_attr, element_name)
+        )
+        try:
+            return response["results"][0]["id"]
+        except (IndexError, KeyError):
+            self.logger.info(
+                "Element '%s' is not declared under foreman collection %s",
+                element_name, self.collection_name
+            )
+            return ""
+
+    def add(self, **kwargs):
+        """
+        Add new element to foreman collection
+
+        :param kwargs: element parameters
+        :raise: ForemanException
+        """
+        try:
+            self.collection.create(kwargs)
+        except Exception as ex:
+            raise ForemanException(
+                "Failed to add element with parameters %s: %s" % (kwargs, ex)
+            )
+
+    def update(self, element_name, **kwargs):
+        """
+        Update element in foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :param kwargs: element parameters to update
+        :raise: ForemanException
+        """
+        element_id = self.get_element_id(element_name=element_name)
+        try:
+            self.collection.update(element_id, kwargs)
+        except Exception as ex:
+            raise ForemanException(
+                "Failed to update element %s with parameters %s: %s" %
+                (element_name, kwargs, ex)
+            )
+
+    def remove(self, element_name):
+        """
+        Remove element from foreman collection
+
+        :param element_name: element name
+        :type element_name: str
+        :raise: ForemanException
+        """
+        element_id = self.get_element_id(element_name=element_name)
```

Comment:
it more general functions, like you can remove any element under specific collection and not a single one